src/mscorlib/src/System/String.cs

```diff
@@ -1153,12 +1218,11 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             } 
             else {
                 int sepListCount = sepList.Length;
-                int sepCount = separator.Length;
                 //If they passed in a string of chars, actually look for those chars.
-                fixed (char* pwzChars = &this.m_firstChar, pSepChars = separator) {
+                fixed (char* pwzChars = &this.m_firstChar) {
                     for (int i=0; i< Length && foundCount < sepListCount; i++) {                        
-                        char * pSep = pSepChars;
-                        for( int j =0; j < sepCount; j++, pSep++) {
+                        char* pSep = separators;
+                        for( int j =0; j < separatorsLength; j++, pSep++) {
```

Comment:
NIT: Spacing (I get you want to reduce chrun, but when you are touching the line anyway, I would prefer to fix the whitespace issues.